### PR TITLE
Implemented readuntil in StreamResponse 

### PR DESCRIPTION
**Pull request by WisdomPill**
## What do these changes do?

Added `readuntil` method, heavily mimicking `asyncio.streams` and recycling the code of readline in current `StreamResponse` class. With this change `readline` uses `readuntil` with `\n` as a separator
Same for the tests that just use different separators.

## Are there changes in behavior for the user?

New way to wait for new data

## Related issue number

#4054 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt` This change is inside another PR I submitted #4732 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by WisdomPill** [comment]
I need to check if documentation work is needed or if it is automatic, meanwhile if somebody can clear my mind about this before I check it is welcome

File: aiohttp/streams.py [hunk 1]
```diff
@@ -300,34 +300,41 @@ async def _wait(self, func_name: str) -> None:
...
+        chunk = b''
+        chunk_size = 0
         not_enough = True
 
         while not_enough:
             while self._buffer and not_enough:
                 offset = self._buffer_offset
-                ichar = self._buffer[0].find(b'\n', offset) + 1
-                # Read from current offset to found b'\n' or to the end.
+                ichar = self._buffer[0].find(separator, offset) + 1
```
Comment (socketpair) [inline aiohttp/streams.py:330]: asyncio has much more effective solution. Why not to copy-paste it ?

Comment (WisdomPill) [reply aiohttp/streams.py:330]: I did not want to change much the code.

What do you think should be copied from asyncio implementation?

Comment (asvetlov) [reply aiohttp/streams.py:330]: I think sync the implementation with asyncio would be great!
asyncio doesn't incrementally copy data to the buffer in a loop but calculates boundaries and move bytes once.

**Review by asvetlov — CHANGES_REQUESTED** [review]
I can live with not the-fast-possible implementation but the documentation update is required for merging.

Please update `docs/streams.rst`

File: aiohttp/streams.py (hunk 1 above)
Comment (WisdomPill) [reply aiohttp/streams.py:330]: Okay, I will fix it

**Comment by WisdomPill** [comment]
Yes, okay, this weekend I will have some time to fix it

**Comment by WisdomPill** [comment]
Hello, 

I tried to mimic more the asyncio implementation but it appears to me that it is quite difficult task and a lot of code will be changed.

Simply copying the data from `self._buffer` using boundaries is not possible as `self._buffer` is a deque.
To change this I can try by making `self._buffer` a `bytearray` which seems to me that it will lead to an improvement in performance but will need a lot of code rewrite.

I added the docs by the way.

**Comment by WisdomPill** [comment]
@asvetlov do you think the StreamReader could be written in a better and more performant way inheriting StreamReader from asyncio?

File: docs/streams.rst [hunk 2]
```diff
@@ -70,6 +70,17 @@ Reading Methods
...
+
+   Read until separator, where `separator` is a sequence of bytes.
+
+   If EOF is received, and `separator` was not found, the method will
+   return the partial read bytes.
+
+   If the EOF was received and the internal buffer is empty, return an
+   empty bytes object.
+
+   :return bytes: the given data
```
Comment (asvetlov) [inline docs/streams.rst]:
```suggestion
   empty bytes object.

   .. versionadded:: 3.8 

   :return bytes: the given data
```

File: docs/streams.rst [hunk 3]
```diff
@@ -70,6 +70,19 @@ Reading Methods
...
+
+   Read until separator, where `separator` is a sequence of bytes.
+
+   If EOF is received, and `separator` was not found, the method will
+   return the partial read bytes.
+
+   If the EOF was received and the internal buffer is empty, return an
+   empty bytes object.
+
+   .. versionadded:: 3.8
```
Comment (asvetlov) [inline docs/streams.rst]:
```suggestion
   .. versionadded:: 3.8
```

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
Thanks for the patience!

**Pull request merged**